clients/go/pkg/zbc/oauthCredentialsProvider.go

```diff
@@ -198,24 +210,37 @@ func applyCredentialDefaults(config *OAuthProviderConfig) {
 			config.Cache = cache
 		}
 	}
+
+	if config.Timeout <= time.Duration(0) {
+		config.Timeout = OAuthDefaultRequestTimeout
+	}
 }
 
-func fetchAccessToken(authorizationServerURL string, payload *oauthRequestPayload) (*OAuthCredentials, error) {
-	jsonPayload, err := json.Marshal(payload)
+func (p *OAuthCredentialsProvider) fetchAccessToken(ctx context.Context) (creds *OAuthCredentials, err error) {
+	req, cancel, err := p.buildOAuthRequest(ctx)
 	if err != nil {
 		return nil, err
 	}
+	defer cancel()
 
-	reader := bytes.NewReader(jsonPayload)
-	response, err := http.Post(authorizationServerURL, "application/json", reader)
+	response, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("failed while requesting access token from URL '%s': %w", authorizationServerURL, err)
+		return nil, fmt.Errorf("failed while requesting access token: %w", err)
 	}
 
-	defer response.Body.Close()
+	defer func() {
+		closeErr := response.Body.Close()
+		if closeErr != nil {
+			if err == nil {
+				err = fmt.Errorf("failed while closing response body: %w", closeErr)
```

Comment:
Does it make sense to fail the whole thing if we failed to close but the rest was ok?

Reply:
The alternative also didn't seem right. If we just log it, then maybe it didn't close the response writer and it will never reuse that connection. If it keeps happening, it will leak resources and fail. But maybe if it's more likely that it will be a transient error, we should just log it or keep trying to close the connection

Reply:
What do you mean by reuse the connection? The `DefaultClient` uses a connection pool, and doesn't allow reusing one until the response writer is closed? What are the cases where a writer might fail to close (that would be unrecoverable)?

Reply:
Yes, that's it. I don't know in which cases it might fail but briefly looking at the Close code it seems that it doesn't?